src/Operations.jl

```diff
@@ -1301,7 +1301,11 @@ function sandbox_preserve(ctx::Context, target::PackageSpec, test_project::Strin
     end
     # preserve important nodes
     keep = [target.uuid]
-    append!(keep, collect(values(read_project(test_project).deps)))
+    test_project_deps = read_project(test_project).deps
+    for (pkg, uuid) in test_project_deps
```

Comment:
I guess this is what CI doesn't like. I did this so I could get the new test packages into the manifest but perhaps this adds too much.

Reply:
We can get rid of this block of changes. Any test packages that don't exist in the active manifest will be resolved later on.

Reply:
I originally didn't have it there but then I got:

```
(Pkg) pkg> test EnglishText
┌ Warning: using test/REQUIRE files is deprecated and current support is lacking in some areas
└ @ Pkg.Operations ~/Pkg.jl/src/Operations.jl:1414
    Testing EnglishText
ERROR: KeyError: key "Compat" not found
Stacktrace:
 [1] getindex at ./dict.jl:477 [inlined]
 [2] #54 at /Users/kristoffer/Pkg.jl/src/manifest.jl:209 [inlined]
 [3] _all(::Pkg.Types.var"#54#57"{Dict{String,Bool}}, ::Dict{String,Base.UUID}, ::Colon) at ./reduce.jl:729
 [4] all at ./reduce.jl:724 [inlined]
 [5] destructure(::Dict{Base.UUID,Pkg.Types.PackageEntry}) at /Users/kristoffer/Pkg.jl/src/manifest.jl:209
 [6] write_manifest at /Users/kristoffer/Pkg.jl/src/manifest.jl:227 [inlined]
 [7] (::Pkg.Operations.var"#86#90"{Pkg.Operations.var"#97#98"{Bool,Cmd,Cmd,Nothing,Pkg.Types.Context,Array{String,1},String,Pkg.Types.PackageSpec},Pkg.Types.Context,Pkg.Types.PackageSpec,String,Pkg.Types.Project,String})(::String) at /Users/kristoffer/Pkg.jl/src/Operations.jl:1356
 [8] #mktempdir#18(::String, ::typeof(mktempdir), ::Pkg.Operations.var"#86#90"{Pkg.Operations.var"#97#98"{Bool,Cmd,Cmd,Nothing,Pkg.Types.Context,Array{String,1},String,Pkg.Types.PackageSpec},Pkg.Types.Context,Pkg.Types.PackageSpec,String,Pkg.Types.Project,String}, ::String) at ./file.jl:634
 [9] mktempdir(::Function, ::String) at ./file.jl:632 (repeats 2 times)
 [10] sandbox(::Function, ::Pkg.Types.Context, ::Pkg.Types.PackageSpec, ::String, ::String, ::Pkg.Types.Project) at /Users/kristoffer/Pkg.jl/src/Operations.jl:1325
 [11] #test#96(::Bool, ::Cmd, ::Cmd, ::Nothing, ::typeof(Pkg.Operations.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/kristoffer/Pkg.jl/src/Operations.jl:1488
 [12] #test at ./none:0 [inlined]
 [13] #test#69(::Bool, ::Nothing, ::Cmd, ::Cmd, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/kristoffer/Pkg.jl/src/API.jl:313
 [14] test(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/kristoffer/Pkg.jl/src/API.jl:300
 [15] #test#68 at /Users/kristoffer/Pkg.jl/src/API.jl:294 [inlined]
 [16] test(::Array{Pkg.Types.PackageSpec,1}) at /Users/kristoffer/Pkg.jl/src/API.jl:294
```